### PR TITLE
Fixed player color assignment crashing with large player IDs

### LINKED ISSUE #1691: SetPlayerColor crashes for player IDs above 38(ish)
https://github.com/octgn/OCTGN/blob/c51f51dc9accc337d34f50653c7a526902358d43/octgnFX/Octgn/Play/Player.cs#L370-L373

This line subtracts 18 from the player ID value, presumably to loop through the color list again when it runs out of colors.  If there's more than 38 players though, the ID will still fall out of the range.

Either stick the Idx-18 into a while loop, or do some modulo/division math to make sure the ID value stays within 1-18.

### PR DESCRIPTION
**Pull request by iGemini**
Fixes #1691

### CODE REVIEW

File: octgnFX/Octgn/Play/Player.cs [hunk 1]
```diff
@@ -335,39 +360,18 @@ public PlayerState State
...
-                                     Color.FromRgb(0x66, 0x66, 0x66),
-                                     Color.FromRgb(0xFF, 0x00, 0x00)
-                                 };
-            if (idx == 255)
-                idx = 0;
-            if (idx > 18)
-                idx = idx - 18;
-
-            ActualColor = baseColors[idx];
+            Color playerColor;
```
[the remark this answers is not in the document]
Comment (iGemini) [reply octgnFX/Octgn/Play/Player.cs]: There's really no advantage, it's just shorter. I prefer readability over form.

[the remark this answers is not in the document]
Comment (iGemini) [reply octgnFX/Octgn/Play/Player.cs]: As far as my understanding goes, yeah. (http://cc.davelozinski.com/c-sharp/what-is-the-fastest-conditional-statement)

[the remark this answers is not in the document]
Comment (Gravecorp) [reply octgnFX/Octgn/Play/Player.cs]: The compiler changes both into nearly the same IL code so the only difference is readability. A shorthand is an easy way to set a value and as long as you dont start nesting short hands its fine to use either.
I also prefer the longer code especially when working with multiple variables.
Compiled code has no size vs performance issues so easier to read is easier to maintain.

**Comment by brine** [comment]
I tested this last night and it should be good to roll out.

File: octgnFX/Octgn/Play/Player.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kellyelton) [reply octgnFX/Octgn/Play/Player.cs]: I'm on the same page as Graves. I prefer code that is more verbose. Performance in this case is irrelevant, though the discussion is interesting never the less.

**Pull request merged**